src/worker/activities/local_activities.rs

```diff
@@ -133,8 +141,15 @@ impl LocalActivityManager {
             .outstanding_activity_tasks
             .remove(task_token)?;
         self.semaphore.add_permits(1);
+        self.complete_notify.notify_one();
         Some(info)
     }
+
+    pub(crate) async fn wait_all_finished(&self) {
+        while !self.dat.lock().outstanding_activity_tasks.is_empty() {
```

Comment:
If you're asking if it does, no, it's dropped after evaluating the condition. If you're asking if it should, no, it would prevent outstanding ones completing.